polaris/benchmark/_base.py

```diff
@@ -302,8 +305,12 @@ def _compute_checksum(self):
             hash_fn.update(c.encode("utf-8"))
         for c in sorted(self.input_cols):
             hash_fn.update(c.encode("utf-8"))
-        for m in sorted(self.metrics, key=lambda k: k.name):
-            hash_fn.update(m.name.encode("utf-8"))
+        # We distinguish between Metrics and GroupedMetrics for backwards compatibility.
+        for name in sorted([m.name for m in self.metrics if isinstance(m, Metric)]):
+            hash_fn.update(name.encode("utf-8"))
+        for d in sorted([m.model_dump() for m in self.metrics if isinstance(m, GroupedMetric)]):
+            s = json.dumps(d, sort_keys="True")
+            hash_fn.update(s.encode("utf-8"))
```

Comment:
**Suggestion**
Encode the hash of the metric instance, rather than one of it's attribute. Then implement `__hash__` in the `Metric` class so that it returns a relevant and unique hash string.
```suggestion
        for metric_hash in sorted([hash(m) for m in self.metrics]):
            hash_fn.update(metric_hash.encode("utf-8"))
```

Reply:
This works, but to maintain backwards compatibility it does mean that the hash for a default metric has to be the label.

Reply:
Actually, `hash()` needs to return an int, so this won't work.